### ISSUE TITLE
Rename MethodCall Rector

### ISSUE THREAD

**Issue by shyim** [rectorphp/rector#1210]
The current RenameMethodRector renames also the ClassMethod itself. So this functionality is unuseable when the method is deprecated.

I could make it configureable :) if wanted

**Comment by TomasVotruba** [comment]
The current renaming class method is included, because in most cases you inherit the class and the new method is preffered.

> I could make it configureable :) if wanted 

Sure :+1: 

How do you plan to do it?

**Comment by shyim** [comment]
```yaml
services:
    Rector\Rector\MethodCall\RenameMethodRector:
        disableClassMethod: true
```

Something like this :thinking:

**Comment by TomasVotruba** [comment]
That might be confusing, like double negation.

Maybe sepearte 2 rectors:

- RenameMethodCallRector (only calls)
- RenameMethodRector (calls + names)

**Comment by shyim** [comment]
Okay i will create a PR :)

**Comment by TomasVotruba** [comment]
Hi, I want to review this issues, as I get very confusing feedback from people. They don't know what's the difference between those 2 rules.

I know it's been long time, so if it's not relevant anymore, let me know ;)


The `RenameMethodCallRector` is used in like 8 cases in Rector sets,
while `RenameMethodRector` in 33.

I feel like there should be just 1 rule. I don't recall situation when there was a preference to one or another.

> So this functionality is unuseable when the method is deprecated.

What exactly is unusable on renaming class method too?

**Comment by shyim** [comment]
I would like offer rector to replace deprecated stuff. But when I run the own set in the library the original code gets changed which is marked to remove in next version. 

So as an user for a library it works fine. That was the idea to just limit to method call. So the calls getting changed to the new method, but the old is still available. Hopefully I could explain my point :)

**Comment by TomasVotruba** [comment]
Ah, I see :).

I wonder what would you use that in a library, when in own library it's better to use PHPStorm refactor method name. This *use-Rector-for-coding* approach would make class rename, constant rename and similar rule useless the same way.

Rector should not replace PHPStorm refactoring, but rather automate the boring part.

**Comment by TomasVotruba** [comment]
2 way how to solve this better:

- prevent old method rename, when the new `ClassMethod` already exists in the class - that would prevent renaming
- skip class itself, e.g. when `Command->execute() → executeIO()` will be in the list, it would skip `Command->execute()`, but include the children classes

**Comment by TomasVotruba** [comment]
Both implemented in #3707, this should prevent few troubles :)
